src/system_info/system_info_network.h

```diff
@@ -80,12 +79,28 @@ class SysInfoNetwork : public SysInfoObject {
   std::string active_connection_;
   std::string active_device_;
   guint device_type_;
-#elif defined(TIZEN)
+#else
   bool GetNetworkType();
-  static void OnTypeChanged(connection_type_e type, void* user_data);
-
-  connection_h connection_handle_;
-#endif
+  void GetCellularNetworkType();
+  static void OnNetworkTypeChanged(keynode_t* node, void* user_data);
+  void SendUpdate();
+#if defined(TIZEN_MOBILE)
+  void OnCellularNetworkTypeChanged(keynode_t* node, void* user_data);
+#else
+  GDBusConnection* conn_;
+  const char* modem_path_;
+  guint cellular_network_type_changed_watch_;
+
+  void UpdateCellularNetwork(const gchar* key, GVariant* var_val);
+  static void OnCellularNetworkTypeChanged(GDBusConnection* conn,
+                                           const gchar* sender_name,
+                                           const gchar* object_path,
+                                           const gchar* iface,
+                                           const gchar* signal_name,
+                                           GVariant* parameters,
+                                           gpointer data);
+#endif  // defined(TIZEN_MOBILE)
+#endif  // defined(GENERIC_DESKTOP)
```

Comment:
There is such a lot ifdef in this header. I'd suggest to redesign this into separate headers. _ivi.h, _mobile.h, _desktop.h.  @qjia7 please work with @liyingzh together to achieve this.